libr/core/anal_tp.c

```diff
@@ -75,6 +75,8 @@ static void var_rename(RAnal *anal, RAnalVar *v, const char *name, ut64 addr) {
 
 static void var_retype(RAnal *anal, RAnalVar *var, const char *vname, const char *type, ut64 addr, bool ref, bool pfx) {
 	r_return_if_fail (anal && var && type);
+	// XXX types should be passed without spaces to trim
+	type = r_str_trim_head_ro (type);
```

Comment:
if you do this, you re-introduce the same bug we had before, where you are writing to read-only memory. Please use r_str_trim_dup instead.

Reply:
nope i dont because _ro means read-only